ssh/server/server.go

```diff
@@ -72,29 +73,37 @@ func NewServer(opts *Options, tunnel *httptunnel.Tunnel) *Server {
 	return server
 }
 
+// SessionHandler handles a new SSH session from ShellHub's users to device.
 func (s *Server) SessionHandler(glidersession ssh.Session) {
 	log.WithFields(log.Fields{
 		"target":  glidersession.User(),
 		"session": glidersession.Context().Value(ssh.ContextKeySessionID),
 	}).Info("Handling session request started")
 
+	// exit responds and finish SSH's session when something goes wrong during session handling.
+	// Internal error is the error from Go code, and external one is what going to be send to end user.
 	exit := func(session ssh.Session, internal, external error) {
 		log.WithFields(log.Fields{
 			"internal": internal,
 			"external": external,
 			"session":  glidersession.Context().Value(ssh.ContextKeySessionID),
 		}).Error("Failed to handler the session")
 
+		// finish close the SSH's session.
 		finish := func(session ssh.Session) {
 			if session != nil {
-				session.Close()
+				err := session.Close()
+				if err != nil {
+					log.WithError(err).Error("Failed to finish the session")
```

Comment:
Here, for example, ought to be in the `var` list.

Reply:
I thought, in this case, as this is only an error message and not an error itself, let it hardcoded.

The `var` errors are the only errors what the ShellHub's user receives. The messages inside `log.Error` are exclusively for ShellHub logging and specific for one case, and the user does not have contact with it.

Even though, should we put this messages into variables?

Reply:
Consistency. Put it there.